### PR TITLE
fix: separate integration workflow from go-test-multiplatform.yml

### PR DESCRIPTION
**Pull request by ykim-1**
## 📝 Description

Noticed that integration test was causing build failure on PRs. This PR separates the integration workflow from unit testing  workflow and running the workflow only on push/merge to dev/main

## ✔️ How to Test

https://github.com/ykim-1/packer-plugin-linode/actions/runs/7790383367
Tests passed, Uploading failing is not relevant

## 📷 Preview

**If applicable, include a screenshot or code snippet of this change. Otherwise, please remove this section.**

### CODE REVIEW

File: .github/workflows/go-test-multiplatform.yml [hunk 1]
```diff
@@ -63,93 +63,12 @@ jobs:
...
-      # Upload test results to the bucket
-      - name: Upload test results to bucket
-        run: |
-          filename=$(ls | grep -E '^[0-9]{12}_packer_test_report\.xml$')
-          python3 tod_scripts/test_report_upload_script.py "${filename}"
-        env:
-          LINODE_CLI_OBJ_ACCESS_KEY: ${{ secrets.LINODE_CLI_OBJ_ACCESS_KEY }}
-          LINODE_CLI_OBJ_SECRET_KEY: ${{ secrets.LINODE_CLI_OBJ_SECRET_KEY }}
-
+          go test -race -count 1 ./builder/linode/... -timeout=3m
```
Comment (zliang-akamai) [inline .github/workflows/go-test-multiplatform.yml:74]:
```suggestion
          make unit-test
```
I think we can simplify it with the makefile target

**Comment by zliang-akamai** [comment]
I think we can rebase `dev` to make commit history cleaner, lol

File: .github/workflows/go-test-multiplatform.yml (hunk 1 above)
Comment (ykim-1) [reply .github/workflows/go-test-multiplatform.yml:74]: Yup will make the change

**Review by zliang-akamai — APPROVED** [review]
Looks good to me!

**Review by lgarber-akamai — APPROVED** [review]
LGTM, excellent work! Could you run `make fmt` before running to fix the CI?

**Pull request merged**